src/notifications/settings/initialization.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  subscribeWalletToAllEnabledTopics,
+  unsubscribeWalletFromAllNotificationTopics,
+} from '@/notifications/settings/firebase';
+import { InteractionManager } from 'react-native';
+import { logger, RainbowError } from '@/logger';
+import { removeNotificationSettingsForWallet } from '@/notifications/settings/settings';
+
+type InitializationStateType = {
+  alreadySaved: Map<
+    string,
+    { index: number; settings: WalletNotificationSettings }
+  >;
+  subscriptionQueue: WalletNotificationSettings[];
+  newSettings: WalletNotificationSettings[];
+};
+
+/**
+ Checks if group notification settings are present in storage
+ and adds default values for them if they do not exist.
+ */
+export const addDefaultNotificationGroupSettings = () => {
+  const data = notificationSettingsStorage.getString(WALLET_GROUPS_STORAGE_KEY);
+
+  if (!data) {
+    const defaultSettings = {
+      [NotificationRelationship.OWNER]: true,
+      [NotificationRelationship.WATCHER]: false,
+    };
+    notificationSettingsStorage.set(
+      WALLET_GROUPS_STORAGE_KEY,
+      JSON.stringify(defaultSettings)
+    );
+  }
+};
+
+/**
+ * Adds fresh disabled settings for all wallets that didn't have settings
+ * schedules subscribing to owned wallets that haven't been initialized yet
+ * schedules removing settings for wallets that are no longer prersent but removal failed previously
+ */
+export const initializeNotificationSettingsForAllAddressesAndCleanupSettingsForRemovedWallets = (
+  addresses: AddressWithRelationship[]
+) => {
+  const initializationState = _prepareInitializationState();
+  // Set of wallet addresses we have in app (unrelated to notification settings entries)
+  const walletAddresses = new Set(
+    addresses.map(addressWithRelationship => addressWithRelationship.address)
+  );
+  const removedWalletsThatWereNotUnsubscribedProperly: string[] = [
+    ...initializationState.alreadySaved.keys(),
+  ].filter(address => !walletAddresses.has(address));
+
+  const queue = _prepareSubscriptionQueueAndCreateInitialSettings(
+    addresses,
+    initializationState
+  );
+
+  InteractionManager.runAfterInteractions(() => {
+    _processSubscriptionQueue(queue);
+  });
+
+  if (removedWalletsThatWereNotUnsubscribedProperly.length) {
+    InteractionManager.runAfterInteractions(() => {
+      removedWalletsThatWereNotUnsubscribedProperly.forEach(address => {
+        removeNotificationSettingsForWallet(address);
+      });
+    });
+  }
+};
+
+/**
+ * Adds fresh disabled settings for all wallets that didn't have settings
+ */
+export const initializeNotificationSettingsForAddresses = (
+  addresses: AddressWithRelationship[]
+) => {
+  const initializationState = _prepareInitializationState();
+
+  const queue = _prepareSubscriptionQueueAndCreateInitialSettings(
+    addresses,
+    initializationState
+  );
+
+  InteractionManager.runAfterInteractions(() => {
+    _processSubscriptionQueue(queue);
+  });
+};
+
+// exported for testing only
```

Comment:
You could write a docblock for this i.e. `/** ... */` so it'll show up if someone imports this method.